### DISCUSSION TITLE
Job concurrency group ignored for build matrix

### DISCUSSION THREAD

**Discussion by harriet-serato** [community/community#26774] [Actions]
<p>Hi, I’m trying to limit job concurrency as per this documentation: <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idconcurrency" class="inline-onebox" rel="noopener nofollow ugc">Workflow syntax for GitHub Actions - GitHub Docs</a></p>
<p>I’ve added <code>concurrency: my_group</code> to the job (which runs twice due to different configurations in a build matrix). However, this seems to be ignored, and the two jobs run in parallel. I would like them to run sequentially.</p>
<p>Is the <code>jobs.&lt;job_id&gt;.concurrency</code> not yet supported for build matrices?</p>

**Comment by harriet-serato** [comment, marked as answer]
<p>The site won’t let me edit or delete this question, but I found a solution: the <code>max-parallel</code> attribute for build matrices can prevent jobs from running in parallel. Interestingly, when I removed the build matrix and manually created the two jobs, the concurrency attribute still didn’t prevent them from running in parallel.</p>

**Comment by hicksjacobp** [comment]
<p>I’ve encountered this same issue, but I wouldn’t consider that <code>max-parallel</code> is really a solution.</p>
<p>I’ve both tried multiple explicit jobs in the same workflow and using a matrix, in which I use <code>concurrency</code>. I have the concurrency group labeled such that it reflects a portion of the matrix jobs. I have 3 jobs in the matrix which target an environment, and I have another 3 jobs in the matrix which target a different environment. The two environments can completely work in parallel, but the jobs for a specific environment cannot. And now I need to add a third environment.</p>
<p>Using <code>max-parallel</code> with 1 wastes a lot of time in my case, whereas if <code>concurrency</code> limited only 1 job in the group from running, it would work perfectly.</p>
<p>The only other workaround I can really think of is if I don’t use a matrix for all the environments, but split each environment into it’s own matrix so I can control that each environment only has 1 job running at a time, but jobs in separate environments can run in parallel. The unfortunate aspect of this approach is that it requires a decent amount of duplicate job configuration and steps.</p>

**Reply by yashanand1910** [reply]
I have this exact requirement and running into the same issue! Did you end up solving it in any way?
